### PR TITLE
added `leftNel` and `rightNel` syntax

### PR DESCRIPTION
**Pull request by kailuowang**
since `EitherNel` is a lot more useful with the `Parallel` TC

### CODE REVIEW

File: core/src/main/scala/cats/syntax/either.scala [hunk 1]
```diff
@@ -329,6 +329,31 @@ final class EitherIdOps[A](val obj: A) extends AnyVal {
...
   /** Wrap a value in `Right`. */
   def asRight[B]: Either[B, A] = Right(obj)
+
+  /**
+   * Wrap a value to a left EitherNel
+   *
+   * For example:
+   * {{{
+   * scala> import cats.implicits._, cats.data.NonEmptyList
+   * scala> "Err".leftNel[Int]
```
Comment (LukaJCB) [inline core/src/main/scala/cats/syntax/either.scala]: Whitespace error on the end of this line

File: core/src/main/scala/cats/syntax/either.scala [hunk 2]
```diff
@@ -329,6 +329,31 @@ final class EitherIdOps[A](val obj: A) extends AnyVal {
...
+   */
+  def leftNel[B]: Either[NonEmptyList[A], B] = Left(NonEmptyList.of(obj))
+
+  /**
+   * Wrap a value to a right EitherNel
+   *
+   * For example:
+   * {{{
+   * scala> import cats.implicits._, cats.data.NonEmptyList
+   * scala> 1.rightNel[String]
```
Comment (LukaJCB) [inline core/src/main/scala/cats/syntax/either.scala]: This line also has an extra space

File: core/src/main/scala/cats/syntax/either.scala [hunk 3]
```diff
@@ -329,6 +329,31 @@ final class EitherIdOps[A](val obj: A) extends AnyVal {
...
+   * Wrap a value to a left EitherNel
+   *
+   * For example:
+   * {{{
+   * scala> import cats.implicits._, cats.data.NonEmptyList
+   * scala> "Err".leftNel[Int]
+   * res0: Either[NonEmptyList[String], Int] = Left(NonEmptyList(Err))
+   * }}}
+   */
+  def leftNel[B]: Either[NonEmptyList[A], B] = Left(NonEmptyList.of(obj))
```
Comment (peterneyens) [inline core/src/main/scala/cats/syntax/either.scala]: You could use `NonEmptyList.one` here.

Comment (LukaJCB) [reply core/src/main/scala/cats/syntax/either.scala]: :+1:

Comment (kailuowang) [reply core/src/main/scala/cats/syntax/either.scala]: Thanks! @peterneyens  addressed.

**Pull request merged**